derive_builder_core/src/builder.rs

```diff
@@ -76,6 +81,11 @@ impl<'a> ToTokens for Builder<'a> {
             let builder_vis = self.visibility;
             let builder_ident = self.ident;
             let derives = self.derives;
+            let clone_derive = if self.pattern.requires_clone() || self.initializer_requires_clone {
```

Comment:
Why is this an 'or' here?

Reply:
It's because the builder needs to derive `Clone` when using the mutable or immutable patterns;

When the builder pattern is owned, but at least one field overrides the pattern to be mutable/immutable, the builder itself doesn't have to. At the time the PR was made, I ~~was a noob~~ just catered for making the builder not derive `Clone` when no fields override the pattern and didn't work through the code to make the builder also not derive `Clone`

Reply:
Okay, I think my version handles this correctly then; it requires clone if any field requires it, whether via explicit or implicitly getting the mutable or immutable pattern.